Topshelf.Linux/packages.config

```diff
@@ -15,5 +14,6 @@
 // specific language governing permissions and limitations under the License.
 -->
 <packages>
-	<package id="Topshelf" version="3.2.0" targetFramework="net40" />
+  <package id="Mono.Posix" version="4.0.0.0" targetFramework="net40" />
```

Comment:
I probable missexplained the issue. It does not have to do with JITing or not. The actual binaries of Mono.Posix.dll which cames bundled with mono is not the same as the one distributed in order to be used on MS.NET (ie. there are different conditional compilation flags in use between both Mono.Posix.dll builds).

On mono/linux Mono.Posix.dll comes bundled (and it is available from GAC). On windows you should bundle Mono.Posix.dll within your own application binaries (like for example, by referencing the available NuGet packages from your solution, as you did).

However, when running an actual application which relies on Mono.Posix.dll and which includes a local Mono.Posix.dll binary, it conflicts with the on available at Mono's GAC.. Something which has happened before to a few users of TopShelf.Linux (we used to reference Mono.Posix at some point).

In order to avoid such conflicts, the actual Mono.Posix code required was "transplanted" in MonoHelper.cs as to avoid referencing Mono.Posix.dll